azure_monitor/src/azure_monitor/export/trace/__init__.py

```diff
@@ -52,104 +53,6 @@ def export(self, spans: Sequence[Span]) -> SpanExportResult:
-                status_code = span.attributes["http.status_code"]
-                data.response_code = str(status_code)
-                data.success = 200 <= status_code < 400
-        else:
-            envelope.name = "Microsoft.ApplicationInsights.RemoteDependency"
-            data = protocol.RemoteDependency(
-                name=span.name,
-                id="{:016x}".format(span.context.span_id),
-                result_code=str(span.status.canonical_code.value),
-                duration=utils.ns_to_duration(span.end_time - span.start_time),
-                success=span.status.canonical_code
-                == StatusCanonicalCode.OK,  # Modify based off attributes or Status
-                properties={},
-            )
-            envelope.data = protocol.Data(
-                base_data=data, base_type="RemoteDependencyData"
-            )
-            if span.kind in (SpanKind.CLIENT, SpanKind.PRODUCER):
-                if (
-                    "component" in span.attributes
-                    and span.attributes["component"] == "http"
-                ):
-                    data.type = "HTTP"
-                if "http.url" in span.attributes:
-                    url = span.attributes["http.url"]
-                    # data is the url
-                    data.data = url
-                    parse_url = urlparse(url)
-                    # TODO: error handling, probably put scheme as well
-                    # target matches authority (host:port)
-                    data.target = parse_url.netloc
-                    if "http.method" in span.attributes:
-                        # name is METHOD/path
-                        data.name = (
-                            span.attributes["http.method"]
-                            + "/"
-                            + parse_url.path
-                        )
-                if "http.status_code" in span.attributes:
-                    status_code = span.attributes["http.status_code"]
-                    data.result_code = str(status_code)
-                    data.success = 200 <= status_code < 400
-            else:  # SpanKind.INTERNAL
-                data.type = "InProc"
-                data.success = True
-        for key in span.attributes:
-            # This removes redundant data from ApplicationInsights
-            if key.startswith("http."):
-                continue
-            data.properties[key] = span.attributes[key]
-        if span.links:
-            links = []
-            for link in span.links:
-                operation_id = "{:032x}".format(link.context.trace_id)
-                span_id = "{:016x}".format(link.context.span_id)
-                links.append({"operation_Id": operation_id, "id": span_id})
-            data.properties["_MS.links"] = json.dumps(links)
-        # TODO: tracestate, tags
+        envelope = convert_span_to_envelope(span)
+        envelope.ikey = self.options.instrumentation_key
```

Comment:
How come we can't set the ikey in the `convert` function?

Reply:
ikey is not available everywhere in the SDK, is currently specific to exporters, validation code is already in there, will need to centralize ikey management to be able to add the ikey in some shared function